ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -267,7 +267,13 @@ void defineCurrent(final ConfigDef configDef) {
   }
 
   private static final Collection<CompatibilityBreakingStreamsConfig>
-      COMPATIBILITY_BREAKING_STREAMS_CONFIGS = ImmutableList.of();
+      COMPATIBILITY_BREAKING_STREAMS_CONFIGS = ImmutableList.of(
+          // Turn on optimizations by default, unless the user explicitly disables in config:
+          new CompatibilityBreakingStreamsConfig(
+            StreamsConfig.TOPOLOGY_OPTIMIZATION,
+            StreamsConfig.OPTIMIZE,
+            StreamsConfig.OPTIMIZE)
+  );
```

Comment:
This config was removed as part of https://github.com/confluentinc/ksql/issues/3622. However, this changed the default value used by KSQL from `all` to `none`, which is not what we want.  Hence I've added this back in.

I guess we could set the default some other way...